問題追加 Wildcard Pattern Matching (#1139)
yosupo06/library-checker-problems: string/wildcard_pattern_matching/task.md

Reviewer: 2^19 の代わりに @{param.S_LEN_MAX} を使ってください

## @{keyword.statement}

@{lang.en}
Given strings $S$ and $T$ consisting of lowercase English letters and asterisks (`*`).

> We say that strings $S'$ and $T'$ are matched if they satisfy either

Reviewer 2: nit: either を 3 つ以上の項目に使用しない (ref: https://english.stackexchange.com/a/170419/406451)
```suggestion
We say that strings $S'$ and $T'$ are matched if they satisfy one of:
```

> We say that strings $S'$ and $T'$ are matched if they satisfy one of:

Reviewer: ```suggestion
We say that strings $S'$ and $T'$ are matched if $|S'|=|T'|$ and they satisfy one of:
```

* $S'_i$ = $T'_i$
* $S'_i$ = `*`
* $T'_i$ = `*`

for each $0\leq i<|S'|$.

For each $0\leq i\leq|S|-|T|$, find if $S[i,i+|T|)$ and $T$ are matched.
@{lang.end}
@{lang.ja}
英小文字とアスタリスク（`*`）からなる文字列 $S$, $T$ が与えられます。

> 文字列 $S'$ と $T'$ が各 $0\leq i<|S'|$ について

Reviewer: ```suggestion
文字列 $S'$ と $T'$ が $|S'|=|T'|$ を満たし、各 $0\leq i<|S'|$ について
```

* $S'_i$ =$T'_i$
* $S'_i$ = `*`
* $T'_i$ = `*`

のいずれかを満たす時，$S'$ と $T'$ はマッチすると呼ぶことにします。

> 各 $0\leq i<|S|-|T|$ について $S[i,i+|T|)$ と $T$ がマッチしているかを求めてください。

Reviewer 2: <|S|-|T| ではなく \leq |S|-|T| と思われます (こことこの上)

Author: 99e679c
